backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/bottomnav/BpkBottomNav.kt

```diff
@@ -0,0 +1,138 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package net.skyscanner.backpack.compose.bottomnav
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.BottomNavigation
+import androidx.compose.material.BottomNavigationDefaults
+import androidx.compose.material.BottomNavigationItem
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+
+@Stable
+sealed interface BpkBottomNavItem {
+  val title: String
+  val id: Int
+  val showBadge: Boolean
+}
+
+fun BpkBottomNavItem(
+  title: String,
+  id: Int,
+  showBadge: Boolean,
+  icon: BpkIcon,
+): BpkBottomNavItem =
+  IconBottomNavItem(title, id, showBadge, icon)
+
+fun BpkBottomNavItem(
+  title: String,
+  id: Int,
+  showBadge: Boolean,
+  painter: Painter,
+): BpkBottomNavItem =
+  PainterBottomNavItem(title, id, showBadge, painter)
+
+
+@Composable
+fun BpkBottomNav(
+  onTabClicked: (Int) -> Unit,
+  selectedItemId: Int,
+  modifier: Modifier = Modifier,
+  elevation: Dp = BottomNavigationDefaults.Elevation,
+  tabItems: List<BpkBottomNavItem> = emptyList(),
+) {
+  Box() {
+    BottomNavigation(
+      modifier = modifier,
+      backgroundColor = BpkTheme.colors.surfaceDefault,
+      contentColor = BpkTheme.colors.textSecondary,
+      elevation = elevation,
+    ) {
+      tabItems.forEach { tabItem ->
+        BottomNavigationItem(
+          selected = selectedItemId == tabItem.id,
+          onClick = { onTabClicked(tabItem.id) },
+          icon = {
+            Box {
+              when (tabItem) {
+                is IconBottomNavItem -> BpkIcon(icon = tabItem.icon, contentDescription = null, size = BpkIconSize.Large)
+                is PainterBottomNavItem -> Icon(
+                  modifier = Modifier.height(24.dp),
+                  painter = tabItem.painter,
+                  contentDescription = null
+                )
+              }
+              NotificationDot(Modifier.align(Alignment.BottomEnd))
```

Comment:
can you please ensure that the badge only shows when `showBadge` is true?
also i think the positioning needs to be tweaked a little based on figma. you may need to use the `offset` modifier as we might need negative padding due to the border